core/src/test/java/org/elasticsearch/test/ElasticsearchTestCase.java

```diff
@@ -371,17 +371,25 @@ public static String randomRealisticUnicodeOfCodepointLength(int codePoints) {
         return RandomizedTest.randomRealisticUnicodeOfCodepointLength(codePoints);
     }
 
-    public static String[] generateRandomStringArray(int maxArraySize, int maxStringSize, boolean allowNull) {
+    public static String[] generateRandomStringArray(int maxArraySize, int maxStringSize, boolean allowNull, boolean allowEmpty) {
         if (allowNull && random().nextBoolean()) {
             return null;
         }
-        String[] array = new String[random().nextInt(maxArraySize)]; // allow empty arrays
+        int arraySize = random().nextInt(maxArraySize);
+        if (!allowEmpty && arraySize == 0) {
+            arraySize = 1;
+        }
```

Comment:
I think this should rather be a loop maybe and get assigned to a random number instead of always 1 whan it was 0 but !allowEmpty ?